### PR TITLE
CreateTupleTypeSymbol shouldn’t reset nullability element types to None if elementNullableAnnotations is a default array.

### LINKED ISSUE #40105: CreateTupleTypeSymbol requires you to pass the element nullabilities explicitly
Compilation.CreateTupleTypeSymbol has an overload that has an elementNullableAnnotations parameter that explicitly lets you give the element's top level nullabilities. This was the API that existed prior types having top-level nullability on them directly. Now, if you call this method but don't pass elementNullableAnnotations at all, I would have expected it would use the top-level nullabilities on the symbols. Instead, it seems those are just dropped:

https://github.com/dotnet/roslyn/blob/22e24f64c2d4b269f5f1aeeb1da57c0ae058db61/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L3135

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #40105.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs [hunk 1]
```diff
@@ -6345,28 +6386,69 @@ public void CreateTupleTypeSymbol_ElementTypes_DefaultArgs()
...
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementNames: default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementLocations: default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
+
+            tuple2 = comp.CreateTupleTypeSymbol(elementTypes, elementNullableAnnotations: default);
+            Assert.True(tuple1.Equals(tuple2, TypeCompareKind.ConsiderEverything));
         }
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: >} [](start = 8, length = 1)

Are we testing cases where `elementTypes` has annotations and so does `elementNullableAnnotations`?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: > Are we testing cases where elementTypes has annotations and so does elementNullableAnnotations?

I believe existing tests below sufficiently cover these cases.

---
In reply to: [357434927](https://github.com/dotnet/roslyn/pull/40348#discussion_r357434927) [](ancestors = 357434927)

**Comment by AlekseyTs** [comment]
@cston Do you have more feedback?

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs (hunk 1 above)
Comment (cston) [reply src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: I couldn't find any test where the annotations on `elementTypes` are not `None` and distinct from `elementNullableAnnotations`.

---
In reply to: [357436843](https://github.com/dotnet/roslyn/pull/40348#discussion_r357436843) [](ancestors = 357436843,357434927)

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: > I couldn't find any test where the annotations on elementTypes are not None and distinct from elementNullableAnnotations.

This PR is focused on scenarios where ```elementNullableAnnotations``` is default.

---
In reply to: [357776096](https://github.com/dotnet/roslyn/pull/40348#discussion_r357776096) [](ancestors = 357776096,357436843,357434927)

**Comment by AlekseyTs** [comment]
@cston, @jcouv, @dotnet/roslyn-compiler Please review, need a second sign-off.

**Pull request merged**